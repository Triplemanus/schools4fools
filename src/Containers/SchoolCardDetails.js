import React from 'react';
import { Link } from 'react-router-dom';
import './SchoolCardDetails.scss';
// import PropTypes from 'prop-types';
import { connect } from 'react-redux';

export const SchoolCardDetails = ({ schools, error, returnRoute}) => {


// export const SchoolCardDetails = ({  school_name, address, phone, distance, returnRoute }) => {
  console.log('S_C_D state, schoolId is : ', schools);
  //<Link to={`${returnRoute}`} className='back-btn'>◀ back</Link>
  return (
    <section className="CardDetails">
      <Link to="/schools" className='back-btn'>◀ back</Link>
      <article id="cardDetails-container">
      <h1 id="school-name">School Name:  {schools.schoolList[1].schoolName}</h1>
      <h2 id="school-address">School Address:  {schools.schoolList[1].address.street}</h2>
      <p className='card-phone'>School Phone:  {schools.schoolList[1].phone}</p>
      <p className='card-distance'>Low Grade:  {schools.schoolList[1].lowGrade}</p>
      </article>
    </section>
  )
}

export const mapStateToProps = state => ({
  ...state
});

export default connect(mapStateToProps)(SchoolCardDetails);

// SchoolCardDetails.propTypes = {
//   school_name: PropTypes.string.isRequired,
//   address: PropTypes.string.isRequired,
//   phone: PropTypes.string.isRequired,
//   distance: PropTypes.number,
//   returnRoute: PropTypes.string.isRequired
// }