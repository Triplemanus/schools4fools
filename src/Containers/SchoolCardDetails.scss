#cardDetails-container {
  display: flex;
  flex-flow: row wrap;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  color: rgb(188, 186, 189);
}

#school-name {
  font-size: 20px;
}

.CardDetails {
  align-items: center;
  background-color: rgb(196, 198, 199);
  border-radius: 12px;
  border: 4px solid #544061;
  color: black;
  display: flex;
  flex-direction: column;
  font-size: 1.5vh;
  height: 610px;
  justify-content: space-evenly;
  margin-bottom: 2px;
  opacity: .7;
  width: 470px;
}

.card-low-grade {
  color: black;
  font-size: 14;
}

.card-hi-grade {
  color: black;
  font-size: 14;
}

#school-name {
  color: black;
  font-size: 16;
}

#school-address {
  color: black;
  font-size: 14;
}

#school-address {
  color: black;
  font-size: 14;
}

.back-btn {
  font-size: 2.5em;
  border: none;
  width: 200px;
  border-radius: 5px;
  &:hover {
    color: white;
    background-color: #BC8F8F;
  }
}