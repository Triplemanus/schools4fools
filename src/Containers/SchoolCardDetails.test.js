import React from 'react'
import { shallow } from 'enzyme'
import SchoolCardDetails from './SchoolCardDetails'

describe('SchoolCardDetails', () => {
  it('should match the snapshot with all data passed in correctly', () => {
    let wrapper = shallow(<SchoolCardDetails 
      school_name= "George Washington High"
      address= "6311 Larchwood Dr., Huntington Beach, CA 92647-2320"
      phone= "(714) 894-7212" 
      distance= {20}
      returnRoute="/"
    />)
    expect(wrapper).toMatchSnapshot()
  })
})