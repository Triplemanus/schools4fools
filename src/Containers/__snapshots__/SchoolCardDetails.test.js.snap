// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SchoolCardDetails should match the snapshot with all data passed in correctly 1`] = `
<section
  className="CardDetails"
>
  <Link
    className="back-btn"
    to="/"
  >
    â—€ back
  </Link>
  <article
    id="cardDetails-container"
  >
    <h1
      id="school-name"
    >
      George Washington High
    </h1>
    <h2
      id="address"
    >
      6311 Larchwood Dr., Huntington Beach, CA 92647-2320
    </h2>
    <p
      className="card-phone"
    >
      (714) 894-7212
    </p>
    <p
      className="card-distance"
    >
      20
    </p>
  </article>
</section>
`;
